### ISSUE TITLE
The second pair of parentheses is removed in invoke chains like a()(){}, which causes a compile error

### ISSUE THREAD

**Issue by func-148254-d** [pinterest/ktlint#3016]
## Expected Behavior

Keep the parentheses if they cannot be omitted.
In my opinion a linter should not cause compile errors.

## Observed Behavior

The second pair of parentheses is removed and this causes a compile error.

## Steps to Reproduce

```kotlin
operator fun Int.invoke() = ""

inline operator fun String.invoke(function: () -> Unit) {}

fun main() {
    1()() {}
}
```
Copy and paste and format with ktlint.
You will get `Kotlin: Too many arguments for 'fun Int.invoke(): String'.`
Because the second pair is removed, Kotlin regards the {} as a lambda argument for Int.invoke rather than another invocation.

## Your Environment

* Version of ktlint used: 1.6.0
* Relevant parts of the `.editorconfig` settings: no .editorconfig; ktlint intellij plugin "distract free" mode without extra settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): https://github.com/nbadal/ktlint-intellij-plugin 0.27.1
* Version of Gradle used (if applicable): no gradle
* Operating System and version: Windows 11 Home 24H2 26100.4061

**Comment by paul-dingemans** [comment]
Tnx, for reporting. I will fix it. 

For now can you either suppress the violation on each invocation as follows:
```
    @Suppress("ktlint:standard:unnecessary-parentheses-before-trailing-lambda")
    1()() {}
```
or disable the rule in `.editorconfig`
```
ktlint_standard_unnecessary-parentheses-before-trailing-lambda=disabled
```
